File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        //std::map c;
       // std::map<int, std::string> m;
       std::map <int,int> counter;
        for(int i=0;i<nums.size();i++)
        {
            auto it = counter.find(nums[i]);
         if (it != counter.end()){ return true;}
         else{
            counter[nums[i]]=1;
         }
        }
        return false;
        
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-0.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:# maximum subarray ---> this is kadanes algorithem 
        answer=[0]*(len(nums)+1)
        d=float('-inf')
        for i in range(len(nums)-1,-1,-1):
            answer[i]=max(nums[i],answer[i+1]+nums[i])
            d=max(answer[i],d)
        return d 


        